refactor(#2119): update button and button row to follow atomic design

jamescallumyoung opened the pull request:
# Description

Refactor the `ActionButton` and `ActionButtonRow` components to be atomic, replacing them with cleaner `Button` and `ButtonRow` components.

NOTE: WIP; This PR does not yet handle the integration of these components.

Fixes # (issue)
Part of the ongoing work towards #2119 

---

Some things you might want to mention:

1. Why are you making the change?

The `Button` component no-longer includes a modal. Instead, the modal and button are seperate components. This preserves the actual DOM structure in the components; the modal isn't _inside_ the button in the DOM, so its not in the components.

Also, the button's props have been updated to expose _most_ of the underlying Ant Design button's props. This may not be desired in most components but I though it may be useful for these very low-level "Ant Design wrapper" components. This behaviour can be updated as we continue the Atomic Design updates.

jamescallumyoung committed:
feat: add disableControls storybook util

jamescallumyoung committed:
refactor: update Button component to atomic design

jamescallumyoung committed:
refactor: update ButtonRow component to atomic design

jamescallumyoung commented on web/utils/storybook-utils.ts:
```diff
@@ -0,0 +1,17 @@
+/**
```
This fn is useful when writing stories for components, as it allows fields to be easily ommitted from the controls panel in Storybook.

jamescallumyoung committed:
docs: remove doc description from Button component.

This means Storybook will use the component's JSDoc comments instead. This is better as this still provides documentation for Storybook but also allows for other documentation tools in the future. Some IDEs (inc. VS Code and IntelliJ) will also show these docs, which is a bonus.

jamescallumyoung committed:
fix: expose target prop on Button atom

jamescallumyoung committed:
docs: add deprecation notice to ActionButton

jamescallumyoung committed:
refactor: use Button in place of ActionButton in Content layout

jamescallumyoung committed:
Prettified Code!

jamescallumyoung commented:
ActionButton is deprecated but ActionButtonRow is not as it is still used by FollowButton et al.

jamescallumyoung commented:
@gabek Could you test this out locally too? I'm still not 100% on running locally.

gabek commented:
Looks like the linting and formatting isn't happy even after it changed some stuff via Prettier.

jamescallumyoung committed:
fix: force use of eslint 8.22

eslint 8.23 has a known issue that causes problems in IntelliJ IDEs.

jamescallumyoung committed:
docs: move ButtonRow desc to JSDoc

jamescallumyoung committed:
chore: update eslint rules to allow dev dependencies in .stories files

jamescallumyoung committed:
format: lint

jamescallumyoung commented:
@gabek fixed the linting issue. The linter failed to fix some issues automatically - sometimes eslint just can't 🤷 

I updated the component .eslint file to change a rule: dev dependencies can be imported in the co-located `.stories.tsx` files. This is done because `@storybook` modules are dev dependencies and are required in stories.

I also moved the `ButtonRow` description to a JSDoc comment, as I suggested you do in #2129 .

gabek commented:
When testing It looks like the optional color is getting lost.

Before:
![image](https://user-images.githubusercontent.com/414923/193144763-174eda6b-8779-4e7c-93bd-8a50a07b1d0d.png)

After:
![image](https://user-images.githubusercontent.com/414923/193144312-a189043c-a160-498c-bfb4-3f7ab503b47c.png)

gabek commented on web/components/atomic/atoms/Button/Button.tsx:
```diff
@@ -0,0 +1,30 @@
+export type ButtonProps = Omit<AntButtonProps, 'children' | 'icon' | 'title'> & {
+  text: string;
+  iconSrc?: string;
+  iconAltText?: string;
```
I think color is missed here, and then not used, so that's why the background color is getting lost.
